daemon/src/model/cfd.rs

```diff
@@ -949,22 +963,32 @@ impl Cfd {
     }
 
     /// Given an attestation, find and decrypt the relevant CET.
-    pub fn decrypt_cet(self, attestation: &oracle::Attestation) -> Result<Option<Event>> {
-        anyhow::ensure!(!self.is_final());
+    pub fn decrypt_cet(
+        self,
+        attestation: &oracle::Attestation,
+    ) -> Result<Option<Event>, DecryptCetError> {
+        if self.is_final() {
+            return Err(DecryptCetError::CfdIsFinal);
+        }
 
         let dlc = match self.dlc.as_ref() {
             Some(dlc) => dlc,
             None => {
-                tracing::warn!(order_id = %self.id(), "Handling attestation without a DLC is a no-op");
-                return Ok(None);
+                return Err(DecryptCetError::NoDlc(self.id()));
+            }
+        };
+
+        let cet = match dlc.signed_cet(attestation) {
+            Ok(cet) => cet,
+            Err(e) => {
+                return Err(DecryptCetError::ConstructingSignedCetFailed(e));
             }
         };
 
-        let cet = match dlc.signed_cet(attestation)? {
+        let cet = match cet {
             Ok(cet) => cet,
             Err(e @ IrrelevantAttestation { .. }) => {
-                tracing::debug!("{}", e);
-                return Ok(None);
+                return Err(DecryptCetError::IrrelevantAttestation(e));
```

Comment:
I think we are introducing a bug here!

We will almost certainly hit this because we feed every attestation to every CFD. This needs to remain an `Ok(None)`.

Reply:
what do you think of matching on this error variant at the call site? It seems like an error from the perspective of the caller of `decrypt_cfd` to me.